packages/actor-init-sparql/lib/HttpServiceSparqlEndpoint.ts

```diff
@@ -147,6 +148,40 @@ Options:
     stderr.write(`Server running on http://localhost:${this.port}/sparql\n`);
   }
 
+  /**
+   * Negotiatiate the best mediatype format
+   * @param {module:http.IncomingMessage} request Request object.
+   * @param {{type: string; quality: number}[]} variants Allowed variants sorted in preferred order.
+   */
+  public contentNegotiation(request: http.IncomingMessage,
+    variants: { type: string; quality: number }[]): string {
+    const isValid = request.headers.accept && request.headers.accept !== '*/*';
+    const negotiation = isValid ? require('negotiate').choose(variants, request) : null;
+    let mediaType = negotiation && negotiation.length > 0 ? negotiation[0].type : null;
+
+    // If we have multiple possible negotiated variants, pick the one with the highest quality
+    if (negotiation && negotiation.length > 1) {
+      // Highest achievable quality
+      const quality = negotiation[0].q;
+      // Accumulate the negotiation variants with the highest quality
+      const negotiationsIndexed = negotiation.filter((element: any) => element.q === quality)
+        .reduce((acc: any, element: any) => {
+          acc[element.type] = element.type;
+          return acc;
+        }, {});
+
+      // Pick the first configured variant that has the desired quality.
+      for (const variant of variants) {
+        if (negotiationsIndexed[variant.type]) {
+          mediaType = variant.type;
+          break;
+        }
+      }
+      return mediaType;
```

Comment:
This return can be removed, since going outside the if-block will return the exact same thing.